### PR TITLE
Add method to post process publications from phenotype tables 

### PR DESCRIPTION
**Pull request by dglemos**

### CODE REVIEW

File: scripts/import/post_process_databases_tables.pl [hunk 1]
```diff
@@ -238,6 +240,280 @@ sub get_connection_and_query {
...
+  foreach my $pheno_data (@{$pheno_feature_data}){
+    my $pheno_feat_id = $pheno_data->[0];
+    my $value_pubid = $pheno_data->[1];
+    
+    my @value_pubid_splited = split /,/, $value_pubid;
+
+    # Get publication with same PMID from study table
+    foreach my $pmid (@value_pubid_splited){
+      # PMID already in the db with PMID=25806919
+      next if $pmid eq '25806920';
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/post_process_databases_tables.pl]: I improved the comments (as discussed) to be easier to understand why this pmid is skipped.

File: scripts/import/post_process_databases_tables.pl [hunk 2]
```diff
@@ -238,6 +240,280 @@ sub get_connection_and_query {
...
+sub process_phenotype_feature {
+  my $reg = shift;
+  my $species = shift;
+  my $dba = shift;
+  my $var_ad = shift;
+  my $pub_ad = shift;
+  my $source_ad = shift;
+
+  ## Get studies from phenotype_feature
+  my $attrib_ext_sth = $dba->dbc()->prepare(qq[ select study_id, source_id, external_reference, study_type
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/post_process_databases_tables.pl]: More efficient. Thanks for the suggestion.

File: scripts/import/post_process_databases_tables.pl [hunk 3]
```diff
@@ -238,6 +240,280 @@ sub get_connection_and_query {
...
+  my ($host, $port) = split /\:/, $host_port;
+
+  print STDERR "\t Publications from Phenotype feature:";
+
+  my $reg = 'Bio::EnsEMBL::Registry';
+  $reg->load_registry_from_db(
+    -host => $host,
+    -user => $user,
+    -port => $port,
+    -pass => $pass,
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/post_process_databases_tables.pl]: `pass` is passed in `-pass`, don't think it's needed in host_port.

[the remark this answers is not in the document]
Comment (helensch) [reply scripts/import/post_process_databases_tables.pl]: I was querying that pass was not a parameter to  post_process_publication - using global pass?

[the remark this answers is not in the document]
Comment (ima23) [reply scripts/import/post_process_databases_tables.pl]: This is consistent with `get_connection_query` that gets `$hname` input. `$hname` is the same as `$host_port` which stores `$hostname`. I'm happy for it to stay the same.

[the remark this answers is not in the document]
Comment (ima23) [reply scripts/import/post_process_databases_tables.pl]: This is consistent with get_connection_query that gets $hname input. $hname is the same as $host_port which stores $hostname. I'm happy for it to stay the same.

File: scripts/import/post_process_databases_tables.pl [hunk 4]
```diff
@@ -238,6 +240,281 @@ sub get_connection_and_query {
...
+    my $study_id = $l->[0];
+    my $source_id = $l->[1];
+    my $external_reference = $l->[2];
+    $external_reference =~ s/PMID://;
+    my $study_type = $l->[3];
+
+    # Get publication with same PMID from study table
+    my $publication = $pub_ad->fetch_by_pmid($external_reference);
+
+    next unless !defined($publication);
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/post_process_databases_tables.pl]: I can't see an example where that could happen.

[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/post_process_databases_tables.pl]: I can't see an example where that could happen.

File: scripts/import/post_process_databases_tables.pl [hunk 5]
```diff
@@ -238,6 +240,281 @@ sub get_connection_and_query {
...
+sub get_epmc_data{
+
+  my $id = shift; ## specific part of URL including pmid or pmcid
+
+  my $http = HTTP::Tiny->new();
+
+  return undef unless defined $id && $id =~/\d+/;
+
+  my $xs = XML::Simple->new();
+  my $request = 'https://www.ebi.ac.uk/europepmc/webservices/rest/search?query=ext_id:' . $id . '%20src:med';
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/post_process_databases_tables.pl]: Comment added.

**Pull request merged**